From 0a128b6bc92cbc3315d4618ed8b9da37b2367573 Mon Sep 17 00:00:00 2001
From: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
Date: Wed, 17 Jan 2018 11:17:08 -0500
Subject: [PATCH 4/4] Update: kvm instrumentation for 4.14.14+, 4.9.77+,
 4.4.112+

Starting from 3.14.14, 4.9.77, and 4.4.112, the 3.14, 4.9, and 4.4
stable kernel branches backport a kvm instrumentation change introduced
in 4.15 which affects the prototype of the kvm_mmio event.

Signed-off-by: Mathieu Desnoyers <mathieu.desnoyers@efficios.com>
---
 instrumentation/events/lttng-module/kvm.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/instrumentation/events/lttng-module/kvm.h b/instrumentation/events/lttng-module/kvm.h
index ea63e88..698cc71 100644
--- a/instrumentation/events/lttng-module/kvm.h
+++ b/instrumentation/events/lttng-module/kvm.h
@@ -85,6 +85,9 @@ LTTNG_TRACEPOINT_EVENT(kvm_ack_irq,
 	{ KVM_TRACE_MMIO_WRITE, "write" }
 
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0) \
+	|| LTTNG_KERNEL_RANGE(4,14,14, 4,15,0) \
+	|| LTTNG_KERNEL_RANGE(4,9,77, 4,10,0) \
+	|| LTTNG_KERNEL_RANGE(4,4,112, 4,5,0) \
 	|| LTTNG_KERNEL_RANGE(3,16,52, 3,17,0) \
 	|| LTTNG_KERNEL_RANGE(3,2,97, 3,3,0))
 
-- 
2.11.1

